pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -135,6 +137,23 @@ func (pl *PodTopologySpread) setListers(factory informers.SharedInformerFactory)
 // EventsToRegister returns the possible events that may make a Pod
 // failed by this plugin schedulable.
 func (pl *PodTopologySpread) EventsToRegister(_ context.Context) ([]framework.ClusterEventWithHint, error) {
+	// Node add|delete|update maybe lead an topology key changed,
+	// and make these pod in scheduling schedulable or unschedulable.
+	//
+	// A note about UpdateNodeTaint event:
+	// NodeAdd QueueingHint isn't always called because of the internal feature called preCheck.
+	// As a common problematic scenario,
+	// when a node is added but not ready, NodeAdd event is filtered out by preCheck and doesn't arrive.
+	// In such cases, this plugin may miss some events that actually make pods schedulable.
+	// As a workaround, we add UpdateNodeTaint event to catch the case.
+	// We can remove UpdateNodeTaint when we remove the preCheck feature.
+	// See: https://github.com/kubernetes/kubernetes/issues/110175
+	nodeActionType := framework.Add | framework.Delete | framework.UpdateNodeLabel | framework.UpdateNodeTaint
+	if pl.enableSchedulingQueueHint {
+		// preCheck is not used when QHint is enabled, and hence we can remove UpdateNodeTaint.
+		nodeActionType = framework.Add | framework.Delete | framework.UpdateNodeLabel
```

Comment:
Depending on the `nodeTaintsPolicy`, UpdateNodeTaint matters, no?
https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/

Reply:
Good point! Re-registered `UpdateNodeTaint` and removes the note since it's actually needed regardless of preCheck.

Reply:
Side questions:
is the hint looking into the `nodeTaintsPolicy` to decide whether to requeue? Do we have an integration test for it?